Assets/MixedRealityToolkit/_Core/Devices/OpenVR/GenericOpenVRController.cs

```diff
@@ -135,272 +231,245 @@ protected void UpdateControllerData(XRNodeState State, MixedRealityInteractionMa
 
-            // If our value changed raise it.
-            if (interactionMapping.Changed)
+            // Raise input system events if it is enabled.
+            if (lastState != TrackingState)
             {
-                //Raise input system Event if it enabled
-                InputSystem?.RaisePoseInputChanged(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction, currentControllerData);
+                InputSystem?.RaiseSourceTrackingStateChanged(InputSource, this, TrackingState);
             }
 
-            if (lastState != TrackingState)
+            if (TrackingState == TrackingState.Tracked && lastControllerPose != currentControllerPose)
             {
-                InputSystem?.RaiseSourceTrackingStateChanged(InputSource, this, TrackingState);
+                if (IsPositionAvailable && IsRotationAvailable)
+                {
+                    InputSystem?.RaiseSourcePoseChanged(InputSource, this, currentControllerPose);
+                }
+                else if (IsPositionAvailable && !IsRotationAvailable)
+                {
+                    InputSystem?.RaiseSourcePositionChanged(InputSource, this, currentControllerPosition);
+                }
+                else if (!IsPositionAvailable && IsRotationAvailable)
+                {
+                    InputSystem?.RaiseSourceRotationChanged(InputSource, this, currentControllerRotation);
+                }
+            }
+        }
+
+        protected void UpdatePointerData(MixedRealityInteractionMapping interactionMapping)
+        {
+            // TODO: configure an offset pointer position for each OpenVR Controller?
+            // Update the interaction data source
+            interactionMapping.PoseData = currentControllerPose; // Currently no way to get pointer specific data, so we use the last controller pose.
+
+            // If our value changed raise it.
+            if (interactionMapping.Changed)
+            {
+                // Raise input system Event if it enabled
+                InputSystem?.RaisePoseInputChanged(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction, interactionMapping.PoseData);
             }
         }
 
         /// <summary>
         /// Update the "Spatial Grip" input from the device
         /// </summary>
-        /// <param name="interactionSourceState">The InteractionSourceState retrieved from the platform</param>
         /// <param name="interactionMapping"></param>
         protected void UpdateGripData(MixedRealityInteractionMapping interactionMapping)
         {
-            //Get the current Grip button state
-            var gripButton = ControllerHandedness == Handedness.Left ? Input.GetAxis(VRInputMappings[10]) : Input.GetAxis(VRInputMappings[11]);
-            switch (interactionMapping.AxisType)
+            // Get the current grip button press state
+            var gripButtonPress = Input.GetAxis(interactionMapping.AxisCode);
```

Comment:
Looks like this isn't used anywhere: 
![image](https://user-images.githubusercontent.com/3580640/43094659-6dd377f2-8e68-11e8-9acb-5046df67d21f.png)